plugins/hls-explicit-record-fields-plugin/src/Ide/Plugin/ExplicitFields.hs

```diff
@@ -504,6 +567,24 @@ getRecCons e@(unLoc -> RecordCon _ _ flds)
     mkRecInfo :: LHsExpr GhcTc -> [RecordInfo]
     mkRecInfo expr =
       [ RecordInfoCon realSpan' (unLoc expr) | RealSrcSpan realSpan' _ <- [ getLoc expr ]]
+getRecCons expr@(unLoc -> app@(HsApp _ _ _)) =
+  let fieldss = maybeToList $ getFields app []
+      recInfo = concatMap mkRecInfo fieldss
+  in (recInfo, not (null recInfo))
+  where
+    mkRecInfo :: RecordAppExpr -> [RecordInfo]
+    mkRecInfo appExpr =
+      [ RecordInfoApp realSpan' appExpr | RealSrcSpan realSpan' _ <- [ getLoc expr ] ]
+
+    getFields :: HsExpr GhcTc -> [LHsExpr GhcTc] -> Maybe RecordAppExpr
+    getFields (HsApp _ (unLoc -> (XExpr (ConLikeTc (conLikeFieldLabels -> fls) _ _))) _) _
+      | null fls = Nothing
+    getFields (HsApp _ constr@(unLoc -> (XExpr (ConLikeTc (conLikeFieldLabels -> fls) _ _))) arg) args
```

Comment:
I think you can just guard this case with `| not (null fls)` and then delete the case above, since the fallthrough case will handle it?